src/commonMain/kotlin/main.kt

```diff
@@ -75,9 +75,21 @@ fun main(args: Array<String>) {
 
     applyModeHints(config.modes, hints, vars)
 
-    val runtimes = configureRuntimes(config, configDir, hints, vars)
     val (launchDirectives, configDirectives) = calculateDirectives(config, hints, vars)
 
+    // Declare the global (runtime-agnostic) directives.
+    val globalDirectiveFunctions: DirectivesMap = mutableMapOf(
+        "help" to { _ -> help(exeFile, programName, supportedOptions) },
+        "apply-update" to { _ -> applyUpdate(appDir, appDir / "update") }
+    )
+
+    // Execute the global directives (e.g. applying updates) before configuring
+    // runtimes (e.g. building classpaths)
+    val runtimeDirectives = executeGlobalDirectives(globalDirectiveFunctions,
```

Comment:
OK if we call this variable `nonGlobalDirectives` or `remainingDirectives`? I was initially surprised by this line of code and assumed it was a cut-and-paste typo, since `runtimeDirectives` doesn't match the "global" elsewhere in the expression.

Reply:
Sure! I was waffling between `nonGlobalDirectives` and `runtimeDirectives`